src/http/request.cr

```diff
@@ -27,6 +27,23 @@ class HTTP::Request
     # will have a format like "IP:port", but this format is not guaranteed.
     # Middlewares can overwrite this value.
     #
+    # Example:
+    #
+    # ```
+    # class ForwarderHandler
+    #   include HTTP::Handler
+    #
+    #   def call(context)
+    #     if ip = context.request.headers["X-Real-IP"]? # When using a reverse proxy that guarantees this field.
+    #       context.request.remote_address = Socket::IPAddress.new(ip, 0)
```

Comment:
```suggestion
    #       context.request.remote_address = Socket::IPAddress.new(ip, 80)
```
That's the most likely scenario.

Reply:
`X-RealIP` just transfers the address, the port is completely unknown. But since `Socket::IPAddress` requires a port, it should be `0` and not a guess that might look like a fact.

Reply:
`0` is still a wrong port that will be logged :/

Reply:
The port number is not logged.

And FYI, in addition to what @straight-shoota said, there is another detail about ports that you don't realize:

The connection goes like this:

User [ 12.34.56.78 : 54321 ] -> [ 1.2.3.4 : 80 ] Site

We are logging 12.34.56.78 here, skipping the 54321 (the user's *outgoing* port) because it's totally random, but you're assuming that 12.34.56.78 : 80 is somehow correct.

Note also that port number 0 is used to indicate "random unused port" which is kinda fitting.

Reply:
Fine. Yes, that's the outgoing port that is usually logger in reverse proxies, it make more sense to log it there than the end application. I still thing weird to have to provide a port in `IPAddress` when it is not always needed, for example calculating masks or in this case.